motoman_driver/src/industrial_robot_client/joint_trajectory_action.cpp

```diff
@@ -532,18 +554,19 @@ void JointTrajectoryAction::controllerStateCB(
     return;
   }
 
-  if (!industrial_utils::isSimilar(all_joint_names_, msg->joint_names))
+  // joint torso_joint_b2 is not in msg->joint_names, so this code is commented out.
```

Comment:
Got it.
But for SDA-series robot (4-groups), this problem is still there, (all_joint_names_ isn't consistent with _msg->joint_names), and the joint_name checking or goal-constraint checking fails. The same with the issue below.
Maybe we can only do this for SDA-series robots?